src/sdk/core/signingcosmwasmclient.ts

```diff
@@ -0,0 +1,805 @@
+    const cometClient = await connectComet(endpoint)
+    return NibiSigningCosmWasmClient.createWithSigner(
+      cometClient,
+      signer,
+      options
+    )
+  }
+
+  /**
+   * Creates an instance from a manually created Comet client.
+   * Use this to use `Comet38Client` or `Tendermint37Client` instead of `Tendermint34Client`.
+   */
+  public static async createWithSigner(
+    cometClient: CometClient,
+    signer: OfflineSigner,
+    options: NibiSigningCosmWasmClientOptions = {}
+  ): Promise<NibiSigningCosmWasmClient> {
+    return new NibiSigningCosmWasmClient(cometClient, signer, options)
+  }
+
+  /**
+   * Creates a client in offline mode.
+   *
+   * This should only be used in niche cases where you know exactly what you're doing,
+   * e.g. when building an offline signing application.
+   *
+   * When you try to use online functionality with such a signer, an
+   * exception will be raised.
+   */
+  public static async offline(
+    signer: OfflineSigner,
+    options: NibiSigningCosmWasmClientOptions = {}
+  ): Promise<NibiSigningCosmWasmClient> {
+    return new NibiSigningCosmWasmClient(undefined, signer, options)
+  }
+
+  protected constructor(
+    cometClient: CometClient | undefined,
+    signer: OfflineSigner,
+    options: NibiSigningCosmWasmClientOptions
+  ) {
+    super(cometClient, options)
+    const {
+      registry = new Registry([...defaultStargateTypes, ...wasmTypes]),
+      aminoTypes = new AminoTypes({
+        ...createDefaultAminoConverters(),
+        ...createWasmAminoConverters(),
+      }),
+    } = options
+    this.registry = registry
+    this.aminoTypes = aminoTypes
+    this.signer = signer
+    this.broadcastTimeoutMs = options.broadcastTimeoutMs
+    this.broadcastPollIntervalMs = options.broadcastPollIntervalMs
+    this.gasPrice = options.gasPrice
+  }
+
+  public async simulate(
+    signerAddress: string,
+    messages: readonly EncodeObject[],
+    memo: string | undefined
+  ): Promise<number> {
+    const anyMsgs = messages.map((m) => this.registry.encodeAsAny(m))
+    const accountFromSigner = (await this.signer.getAccounts()).find(
+      (account) => account.address === signerAddress
+    )
+    if (!accountFromSigner) {
+      throw new Error("Failed to retrieve account from signer")
+    }
+    const pubkey = encodeSecp256k1Pubkey(accountFromSigner.pubkey)
+    const { sequence } = await this.getSequence(signerAddress)
+    const { gasInfo } = await this.forceGetQueryClient().tx.simulate(
+      anyMsgs,
+      memo,
+      pubkey,
+      sequence
+    )
+    assertDefined(gasInfo)
+    return Uint53.fromString(gasInfo?.gasUsed.toString()).toNumber()
+  }
+
+  /** Uploads code and returns a receipt, including the code ID */
+  public async upload(
+    senderAddress: string,
+    wasmCode: Uint8Array,
+    fee: FeeOption,
+    memo = "",
+    instantiatePermission?: AccessConfig
+  ): Promise<UploadResult> {
+    const compressed = pako.gzip(wasmCode, { level: 9 })
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/Compression_Streams_API

Reply:
https://gist.github.com/Explosion-Scratch/357c2eebd8254f8ea5548b0e6ac7a61b

Reply:
Idk, I'd rather stick to this library and rely on the cosmjs choices.
They don't have enough context but they have made the choice on pako for maintainability.

https://github.com/cosmos/cosmjs/issues/113